cache/cache.go

```diff
@@ -12,39 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package cache is a simple, local, in-memory key-value store.
+// Package cache is a simple, local, goroutine-safe in-memory key-value store.
 package cache
 
 import "sync"
 
-// Object represents a single object in the cache. Although this could be represented
-// as a bare interface{}, we chose to create a new Object type in case we wish to add
-// additional functionality or metadata in the future.
-type Object struct {
-	Contents interface{}
+// Cache represents the interface and available methods of the dcos-go/cache package.
+// At the moment, the only implementation is SimpleCache(), although this could be
+// expanded in the future to handle additional scenarios.
+type Cache interface {
+	Delete(string)
+	Get(string) (interface{}, bool)
+	Objects() map[string]interface{}
+	Purge()
+	Set(string, interface{})
+	Size() int
+	Supplant(map[string]interface{})
 }
 
-// Cache represents the structure of the objects in the cache (objects)
-// and a single locking mechanism (mutex) that is shared across a given instance
-// of the cache.
-type Cache struct {
-	objects map[string]Object
+// cacheImpl represents the structure of the cache, including the cache objects
+// and a single locking mechanism that is shared across a given instance, ensuring
+// some level of goroutine-safety.
+type cacheImpl struct {
+	objects map[string]object
 	mutex   sync.RWMutex
 }
 
-// New creates a new cache and returns it to the caller. The caller is then
-// responsible for creating and deleting objects.
-func New() *Cache {
-	return &Cache{objects: make(map[string]Object)}
+// object represents a single object in the cache. Although this could be represented
+// as a bare interface{}, we chose to create a new object type in case we wish to add
+// additional functionality or metadata in the future.
+type object struct {
+	contents interface{}
+}
+
+// SimpleCache creates a new, basic, in-memory cache. The caller must handle
+// all Set() and Delete() operations by itself; that is to say, there is no
+// concept of a maximum size or expiration on cached objects.
+func SimpleCache() Cache {
```

Comment:
nit: name stutter. people will call this like `foo := cache.SimpleCache()`. consider renaming? perhaps `New()` would suffice (`foo := cache.New()` reads better)?